theano/sandbox/cuda/blas.py

```diff
@@ -39,79 +42,83 @@ def c_code(self, node, name, input_names, output_names, sub):
         bx, by = input_names
         bz, = output_names
         fail = sub['fail']
-        return """
-        float alpha = 1.0;
-        float beta = 0.0;
-
-        int i, x_dim0, x_dim1, x_dim2, y_dim0, y_dim1, y_dim2;
-        int x_stride, y_stride, z_stride, total_size;
-        int ptr_array_size = 3 * CudaNdarray_HOST_DIMS(%(bx)s)[0] * sizeof(float *);
-        int out_dim[3];
-
-        cublasStatus_t err;
-        cudaError_t err1;
-
-        float **host_x = NULL;
-        float **host_z = NULL;
-        float **host_y = NULL;
-
-        float **gpu_x = NULL;
-        float **gpu_y = NULL;
-        float **gpu_z = NULL;
+        threshold = self.stream_threshold
+        return ("""
+        float alpha = 1.0, beta = 0.0;
 
-        x_dim0 = CudaNdarray_HOST_DIMS(%(bx)s)[0];
-        x_dim1 = CudaNdarray_HOST_DIMS(%(bx)s)[1];
-        x_dim2 = CudaNdarray_HOST_DIMS(%(bx)s)[2];
+        const int* Nx = CudaNdarray_HOST_DIMS(%(bx)s);
+        const int* Ny = CudaNdarray_HOST_DIMS(%(by)s);
+        int Nz[3] = {0};
 
-        y_dim0 = CudaNdarray_HOST_DIMS(%(by)s)[0];
-        y_dim1 = CudaNdarray_HOST_DIMS(%(by)s)[1];
-        y_dim2 = CudaNdarray_HOST_DIMS(%(by)s)[2];
+        // use parallel cublasSgemm calls rather than cublasSgemmBatched for large products
+        // (compute products in double because they can be large and we don't need to be exact)
+        bool use_cublas_sgemm_batched = (
+            double(Nx[1]) * double(Nx[2]) * double(Nx[2]) <
```

Comment:
Actually I intended Nx[1], Nx[2], Ny[2]. It's the number of multiplications per instance in the batch.